components/apimgt/org.wso2.carbon.apimgt.rest.api.commons/src/main/java/org/wso2/carbon/apimgt/rest/api/common/interceptors/RESTAPISecurityInterceptor.java

```diff
@@ -118,6 +119,15 @@ public boolean preCall(Request request, Response response, ServiceMethodInfo ser
                 response.setStatus(javax.ws.rs.core.Response.Status.OK.getStatusCode()).setEntity(Json.pretty
                         (swagger)).setMediaType(MediaType.APPLICATION_JSON).send();
                 return false;
+            } else if (requestURI.contains("swagger.yaml")) {
+                try {
+                    yamlContent = RestApiUtil.getStoreRestAPIResource();
+                } catch (APIManagementException e) {
+                    log.error("Couldn't find swagger.yaml for store", e);
+                }
+                response.setStatus(javax.ws.rs.core.Response.Status.OK.getStatusCode()).setEntity(yamlContent)
```

Comment:
@tmkasun We have to send 404 if the resource "/api/am/store/v1.0/swagger.yaml" is not available in the API. But we are claiming that it is there. So IMHO, we can't say we don't have that resource by sending 404. If we can't find that file to serve in our system, I believe its a problem in the implementation. If we send 404 to the client, the client gets the impression that he is using a wrong URL which is not the case.